homeassistant/components/owlet/sensor.py

```diff
@@ -0,0 +1,186 @@
+"""Support for Owlet sensors."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    PERCENTAGE,
+    SIGNAL_STRENGTH_DECIBELS_MILLIWATT,
+    UnitOfTemperature,
+    UnitOfTime,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+
+from .const import DOMAIN, SLEEP_STATES
+from .coordinator import OwletCoordinator
+from .entity import OwletBaseEntity
+
+
+@dataclass
+class OwletSensorEntityDescriptionMixin:
+    """Owlet sensor description mix in."""
+
+    element: str
+
+
+@dataclass
+class OwletSensorEntityDescription(
+    SensorEntityDescription, OwletSensorEntityDescriptionMixin
+):
+    """Represent the owlet sensor entity description."""
+
+
+SENSORS: tuple[OwletSensorEntityDescription, ...] = (
+    OwletSensorEntityDescription(
+        key="batterypercentage",
+        translation_key="batterypercent",
+        native_unit_of_measurement=PERCENTAGE,
+        device_class=SensorDeviceClass.BATTERY,
+        state_class=SensorStateClass.MEASUREMENT,
+        element="battery_percentage",
+    ),
+    OwletSensorEntityDescription(
+        key="oxygensaturation",
+        translation_key="o2saturation",
+        native_unit_of_measurement=PERCENTAGE,
+        state_class=SensorStateClass.MEASUREMENT,
+        element="oxygen_saturation",
+        icon="mdi:leaf",
+    ),
+    OwletSensorEntityDescription(
+        key="oxygensaturation10a",
+        translation_key="o2saturation10a",
+        native_unit_of_measurement=PERCENTAGE,
+        state_class=SensorStateClass.MEASUREMENT,
```

Comment:
This is an average, which is not a current measurement

```suggestion
```

Reply:
Please implement the requested change